doc/specification.md

```diff
@@ -122,11 +122,11 @@ An `AFTResult` message MUST have the followings fields populated by the device:
 
 In `RIB_AND_FIB_ACK` acknowledge mode, it's possible that a gRIBI entry is installed in the RIB, but is not the preferred route (e.g., there is a static route for the same matching entry), and therefore the gRIBI entry will not be programmed into the FIB. In this case, the device should only respond with the `status` value `RIB_PROGRAMMED`.
 
-##### [TODO] 4.1.3.3.1 Idempotent ADD and REPLACE
+##### 4.1.3.3.1 Idempotent DELETE
 
-Clarify the following scenarios:
-* An entry is already installed in the FIB, received an AFTOperation for adding the same entry.
-* An entry was failed to be programmed into the FIB, received an AFTOperation for adding the same entry.
+The behavior of AFT operation `DELETE` MUST be idempotent as to the device RIB/FIB state. In addition, the idempotent behavior should also cover the response. For example, if the entry does not exist, the device should return `FIB_PROGRAMMED` (in the session of `ack_type=RIB_AND_FIB_ACK`).
+
+It is normal and expected that controllers might send repeated `DELETE`, or send a `DELETE` while one is still pending processing on the device. Having this behavior simplifies the implementation, instead of overloading `FAILED` or disconnecting the `Modify` RPC with errors. If future use cases/issues say different, we can consider enrich the `AFTResult` message.
```

Comment:
```suggestion
It is normal and expected that controllers might send repeated `DELETE`, or send a `DELETE` while one is still pending processing on the device. Having this behavior simplifies the implementation, instead of overloading `FAILED` or disconnecting the `Modify` RPC with errors.
```

Suggest to lose the last sentence, since this is pretty much true for anything about the spec :)